crates/net/src/builder.rs

```diff
@@ -111,12 +111,18 @@ impl NetworkDriverBuilder {
         let behaviour = Behaviour::new(config, &[Box::new(handler.clone())])?;
 
         // Build the swarm.
+        let noise_config = self.noise_config.take();
         let keypair = self.keypair.take().unwrap_or(Keypair::generate_secp256k1());
         let swarm = SwarmBuilder::with_existing_identity(keypair)
             .with_tokio()
-            .with_tcp(self.tcp_config.take().unwrap_or_default(), NoiseConfig::new, || {
-                self.yamux_config.take().unwrap_or_default()
-            })?
+            .with_tcp(
+                self.tcp_config.take().unwrap_or_default(),
+                |i: &Keypair| match noise_config {
+                    Some(cfg) => core::result::Result::Ok(cfg),
```

Comment:
why can't we just do Ok(cfg)? I'm guessing the net crate is no_std by default?

Reply:
Good point! This was a relic of debugging an issue with specifying the argument type as `&Keypair`, which fixed the compiler not knowing the signature of the closure. Fixed now 👍